packages/zoe/test/swingsetTests/brokenContracts/vat-alice.js

```diff
@@ -0,0 +1,469 @@
+    });
+    const refundPayments = { Asset: refundPayment };
+
+    const { payout: payoutP, outcome: refundOutcome } = await E(zoe).offer(
+      refundInvite,
+      refundProposal,
+      refundPayments,
+    );
+
+    // The swap deposit should be refunded
+    await showPurseBalance(moolaPurseP, 'aliceMoolaPurse', log);
+    await showPurseBalance(simoleanPurseP, 'aliceSimoleanPurse', log);
+
+    refundOutcome.then(
+      () => assert(false, ' expected outcome to fail'),
+      e => log(`outcome correctly resolves to broken: ${e}`),
+    );
+
+    payoutP.then(async refundPayout => {
+      const { Asset: swapMoolaPayout, Price: swapSimoleanPayout } = E.G(
+        refundPayout,
+      );
+      E(moolaPurseP).deposit(await swapMoolaPayout);
+      E(simoleanPurseP).deposit(await swapSimoleanPayout);
+
+      showPurseBalance(moolaPurseP, 'aliceMoolaPurse', log);
+      showPurseBalance(simoleanPurseP, 'aliceSimoleanPurse', log);
+      E(moolaPurseP)
+        .getCurrentAmount()
+        .then(amount => console.log(`refund value, ${amount.value}`));
+    });
+
+    E(publicAPI)
+      .getOffersCount()
+      .then(
+        () => assert(false, 'contract should be non-responsive '),
+        e => log(`contract no longer responds: ${e}`),
+      );
+
+    // zoe should still be able to make new vats.
+    const { instanceRecord: newInstanceRecord } = await E(zoe).makeInstance(
+      installId,
+      issuerKeywordRecord,
+    );
+    logCounter(log, newInstanceRecord.publicAPI);
+  };
+
+  const doThrowInHook = async () => {
+    log(`=> alice.doThrowInHook called`);
+    const installId = installations.crashAutoRefund;
+
+    const issuerKeywordRecord = harden({
+      Asset: moolaIssuer,
+      Price: simoleanIssuer,
+    });
+    const { instanceRecord } = await E(zoe).makeInstance(
+      installId,
+      issuerKeywordRecord,
+    );
+    const proposal = harden({
+      give: { Asset: moola(3) },
+      want: { Price: simoleans(7) },
+      exit: { onDemand: null },
+    });
+    const alicePayments = { Asset: moolaPayment };
+
+    logCounter(log, instanceRecord.publicAPI);
+    const invite = await E(instanceRecord.publicAPI).makeThrowingInvite();
+    const { payout: payoutP, outcome } = await E(zoe).offer(
+      invite,
+      proposal,
+      alicePayments,
+    );
+
+    outcome.then(
+      () => assert(false, ' expected outcome to fail'),
+      e => log(`outcome correctly resolves to broken: ${e}`),
+    );
+    logCounter(log, instanceRecord.publicAPI);
+    const payout = await payoutP;
+    const moolaPayout = await payout.Asset;
+    const simoleanPayout = await payout.Price;
+
+    await E(moolaPurseP).deposit(moolaPayout);
+    await E(simoleanPurseP).deposit(simoleanPayout);
+    await showPurseBalance(moolaPurseP, 'aliceMoolaPurse', log);
+    await showPurseBalance(simoleanPurseP, 'aliceSimoleanPurse', log);
+    logCounter(log, instanceRecord.publicAPI);
+
+    // zoe should still be able to make new vats.
```

Comment:
Hmm, an offerHandler throwing shouldn't affect anything other than the particular offer. Do we test this? It looks like we are assuming it kills the instance, but we should specifically check that it doesn't.

Reply:
Hmm, isn't this function called 'doThrowInHook'? I think you added the `newInvite` since I looked at it last, so thank you for doing that! That addresses my concerns about ensuring the instance is still up after an offerHandler throws.